addon/components/navigation-narrator.js

```diff
@@ -88,8 +105,22 @@ export default class NavigationNarratorComponent extends Component {
 
   @action
   onRouteChange(transition) {
-    let shouldFocus = this.routeChangeValidator(transition);
+    let shouldFocus;
+    this.transition = transition; // We need to do this because we can't pass an argument to a getter
+
+    if (this.excludeAllQueryParams === true) {
+      if (this.hasQueryParams === true) {
+        return;
+      } else {
+        // it does not matter, there aren't any query params
+        shouldFocus = this.routeChangeValidator(transition);
+      }
+    } else {
+      // excludeAllQueryParams is false, carry on
+      shouldFocus = this.routeChangeValidator(transition);
+    }
```

Comment:
Minor potential refactor here:

```suggestion
    if (this.excludeAllQueryParams && this.hasQueryParams) {
      return;
    }

    shouldFocus = this.routeChangeValidator(transition);
```